pkg/coveragedb/coveragedb.go

```diff
@@ -627,3 +622,58 @@ func UniqCoverage(fullCov, partCov map[int]int64) map[int]int64 {
 	}
 	return res
 }
+
+func RegenerateSubsystems(ctx context.Context, ns string, sss []*subsystem.Subsystem,
+	client spannerclient.SpannerClient) (int, error) {
+	ssMatcher := subsystem.MakePathMatcher(sss)
+	ssCache := make(map[string][]string)
+	filePaths, err := getFilePaths(ctx, ns, client)
+	if err != nil {
+		return 0, err
+	}
+	var mutations []*spanner.Mutation
+	for _, filePath := range filePaths {
+		subsystems := getFileSubsystems(filePath, ssMatcher, ssCache)
+		mutations = append(mutations, fileSubsystemsMutation(ns, filePath, subsystems))
+	}
+	// There is a limit on the number of mutations per transaction (80k) imposed by the DB.
+	// Expected mutations count is < 20k and looks safe to do w/o batching.
+	if _, err = client.Apply(ctx, mutations); err != nil {
+		return 0, err
+	}
+	return len(mutations), nil
+}
+
+func getFilePaths(ctx context.Context, ns string, client spannerclient.SpannerClient) ([]string, error) {
+	iter := client.Single().Query(ctx, spanner.Statement{
+		SQL: `select
```

Comment:
Nit: not sure that splitting this query into multiple lines and having so many tabs helps its understanding.

Reply:
Fixed.